src/actions.ts

```diff
@@ -12,7 +12,7 @@ function getActionName(action: string) {
 class ActionManager implements Disposable {
     private disposables: Disposable[] = [];
     private actions: string[] = [];
-    private client!: NeovimClient;
+    private client?: NeovimClient;
```

Comment:
This is an example of why `!` should rarely be used. It is used too often in this codebase.

P.S. where is the code that actually initializes this `client` ?

Reply:
In fact, within the nvim-related logic, the client is always instantiated. However, this is indeed a special case because this event is registered when the Action is instantiated.

Reply:
> the client is always instantiated

where?

Reply:
Why are we doing that? The comment above it gives no insight.